aws/bucket.ts

```diff
@@ -16,6 +16,10 @@ import * as aws from "@pulumi/aws";
 import * as cloud from "@pulumi/cloud";
 import * as pulumi from "@pulumi/pulumi";
 
+import * as callback from "./callback";
+
+export type BucketHandler = callback.AwsCallback<(args: cloud.BucketHandlerArgs) => Promise<void>>;
```

Comment:
aws/azure specific libraries now say they take an **AwsCallback** or **AzureCallback**.  This indicate they can either take a simple JavaScript function, or they can take the richer cloud-provider-specific object that ghe Lambda/FunctionApp will be created out of.